hilla: add an article to explain how to cancel endpoint requests (#1514)
vaadin/docs: articles/advanced/cancelling-endpoint-requests.asciidoc

---
title: Cancelling Endpoint Requests
description: Hilla endpoint requests can be cancelled using standard AbortControllers.
order: 200
layout: page
---

[role="since:com.vaadin:vaadin@V1.2"]
== Cancelling Endpoint Requests

Sometimes it can be useful to programmatically cancel some requests, for example when they are taking too long or are waiting in a queue despite no longer being needed.
Hilla allows developers to cancel these requests using standard https://developer.mozilla.org/en-US/docs/Web/API/AbortController[AbortController]s.

> All endpoint requests support an optional `init` parameter that can be used to pass an `AbortSignal` which will allow them to be cancelled.

Reviewer: Should probably mention the new name, or remove the name entirely. It’s not much useful anyway.

```suggestion
All endpoint requests support an optional last object parameter, that can be used to pass an `AbortSignal`, which will allow them to be cancelled.
```

Author: Good idea. The name can even change when using the multi-module generator. What if I skip it altogether?

```suggestion
All endpoint requests support an optional parameter that can be used to pass an `AbortSignal` which will allow them to be cancelled.
```

---
title: Cancelling endpoint requests

> description: TODO

Reviewer 2: Let's remember to add the description before merging.

order: 200
layout: page
---

[role="since:com.vaadin:vaadin@V1.2"]
== Cancelling endpoint requests

Sometimes it can be useful to programmatically cancel some requests, for example when they are taking too long

> or they are standing in queue despito no longer being needed. Hilla allows developers to cancel those

Reviewer 2: ```suggestion
or they are standing in queue despite no longer being needed. Hilla allows developers to cancel those
```

request using standard https://developer.mozilla.org/en-US/docs/Web/API/AbortController[AbortController]s.

> All endpoint requests support an optional `init` parameter that can be used to pass an `AbortSignal` which will
> allow to cancel them.

Reviewer 3: ```suggestion
All endpoint requests support an optional `init` parameter that can be used to pass an `AbortSignal` which will allow them to be cancelled.
```
